tests/network/checkup_framework/utils.py

```diff
@@ -256,13 +256,16 @@ def wait_for_job_failure(job):
         job_status = TimeoutSampler(
             wait_timeout=timeout,
             sleep=1,
-            func=lambda: job.instance.status.conditions[0],
+            func=lambda: filter(
+                lambda cond: cond["status"] == job.Condition.Status.TRUE, job.instance.status.conditions
+            ),
         )
         for sample in job_status:
-            if sample["type"] == job.Status.FAILED and sample["status"] == job.Condition.Status.TRUE:
-                return
-            if sample["type"] == job.Status.SUCCEEDED and sample["status"] == job.Condition.Status.TRUE:
-                raise ResourceValueError(f"Job {job.name} has succeeded and should have failed.")
+            for condition in sample:
+                if condition["type"] == job.Status.FAILED and condition["status"] == job.Condition.Status.TRUE:
```

Comment:
`condition["status"] == job.Condition.Status.TRUE` is not needed here anymore.

Reply:
Correct, thank you.